### PR TITLE
feat: cache object pruning

### PR DESCRIPTION
**Pull request by csviri**
Adding possibility to prune objects in the caches.

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/informer/InformerConfiguration.java [hunk 1]
```diff
@@ -202,12 +213,17 @@ public InformerConfigurationBuilder<R> withGenericFilter(GenericFilter<R> generi
       return this;
     }
 
+    public InformerConfigurationBuilder<R> withCachePruneFunction(UnaryOperator<R> itemStore) {
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/informer/InformerConfiguration.java:216]: Why is the parameter called itemStore?

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ControllerConfigurationOverrider.java [hunk 2]
```diff
@@ -158,6 +161,11 @@ public ControllerConfigurationOverrider<R> withGenericFilter(GenericFilter<R> ge
     return this;
   }
 
+  public ControllerConfigurationOverrider<R> withItemStore(UnaryOperator<R> cachePruneFunction) {
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ControllerConfigurationOverrider.java:164]: This should be called `withCachePruneFunction`

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/informer/InformerConfiguration.java (hunk 1 above)
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/informer/InformerConfiguration.java:216]: will rename it, before the whole item store was customizable, later removed it

**Review by metacosm — CHANGES_REQUESTED** [review]

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ControllerConfigurationOverrider.java (hunk 2 above)
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ControllerConfigurationOverrider.java:164]: fixed

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/informer/InformerConfiguration.java (hunk 1 above)
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/informer/InformerConfiguration.java:216]: fixed

**Review by metacosm — APPROVED** [review]

**Pull request merged**